docs: update development-related documentation (#31043)
electron/electron: docs/development/testing.md

its own `package.json` and that its dependencies are therefore not defined
in the top-level `package.json`.

To run only tests in a specific process, run `npm run test --runners=PROCESS`
where `PROCESS` is one of `main` or `remote`.

To run only specific tests matching a pattern, run `npm run test --
-g=PATTERN`, replacing the `PATTERN` with a regex that matches the tests
you would like to run. As an example: If you want to run only IPC tests, you
would run `npm run test -- -g ipc`.

## Node.js Smoke Tests

If you've made changes that might affect the way Node.js is embedded into Electron,

> we have a test runner that runs all of Node.js' own tests with Electron's custom fork

Reviewer: Nit: I prefer to avoid possessives with words that end in `s`
```suggestion
we have a test runner that runs all of the tests from Node.js, using Electron's custom fork
```

of Node.js.

> To run all of Node.js' tests:

Reviewer: Nit
```suggestion
To run all of the Node.js tests:
```
